ASG EC2 Auto EIP

igorkotof opened the pull request.

Commit:
ASG EC2 Auto EIP

DimmKirr commented on data.tf:
```diff
@@ -9,5 +9,11 @@ data "template_file" "asg_ecs_ec2_user_data" {
     ecs_cluster_name  = local.ecs_cluster_name
     env               = var.env
     ec2_service_group = var.ec2_service_group
+    auto_assign_eip   = var.ec2_asg_auto_assign_eip
```
Can we use https://www.terraform.io/language/functions/tostring here and switch the whole var to bool?

DimmKirr commented on autoscaling.tf:
```diff
@@ -55,3 +65,23 @@ module "autoscaling" {
+      },
+    ]
+  })
+}
```
I'd prefer a new line here please

DimmKirr commented on ecs_ec2_user_data.sh.tpl:
```diff
@@ -45,3 +45,110 @@ echo "*/5 * * * * root /etc/ecs/cw_ecsmetrics.sh" > /etc/cron.d/ecsmetrics
+TIMEOUT=20
+PAUSE=5
+
+apt-get update
```
As far as I know AWS ECS optimized AMI is is RHEL based, any reason to use ubuntu for that?

DimmKirr commented on ecs_ec2_user_data.sh.tpl:
```diff
@@ -45,3 +45,110 @@ echo "*/5 * * * * root /etc/ecs/cw_ecsmetrics.sh" > /etc/cron.d/ecsmetrics
+apt install -y curl awscli
+
+aws_get_instance_id() {
+	instance_id=$( (curl http://169.254.169.254/latest/meta-data/instance-id) )
```
Any reason why there are double braces?

DimmKirr commented on autoscaling.tf:
```diff
@@ -1,3 +1,13 @@
+  vpc               = false
+  public_ipv4_pool  = "amazon"
+
+  tags = {
```
Should we add some sequential id here?

igorkotof replied:
it's not needed @AutomationD

DimmKirr commented on variables.tf:
```diff
@@ -366,6 +366,12 @@ variable "autoscale_scheduled_down" {
 }
 
+variable "ec2_asg_auto_assign_eip" {
+  type        = string
```
If we use `tostring()` in the template, we could switch to bool here, I think

DimmKirr reviewed (changes requested):
Looks good, thank you. There are some suggestions though.

Commit:
PR fix

DimmKirr committed:
EC2 ASG IP
- Add ec2_eip_count parameter
- Add EIP name
- Use role_name to asg
- Fix formatting
- Use tostring and bool for consistency
- Terraform template escape $
- Output list of EIPs

Commit:
PR fix 2

Commit:
Merge branch 'asg-eip' of https://github.com/hazelops/terraform-aws-ecs-app into asg-eip

Commit:
Revert "PR fix 2"

This reverts commit c3bfe631a96180b4ceaedc4ba694afa4420e80ba.

DimmKirr committed:
Fix type typo

DimmKirr committed:
Merge pull request #9 from hazelops/feature/asg-eip-fixes

EC2 ASG IP
